bundles/framework/printout/PrintoutHandler.js

```diff
@@ -0,0 +1,317 @@
+import React from 'react';
+import { StateHandler, controllerMixin, Messaging } from 'oskari-ui/util';
+import { Message } from 'oskari-ui';
+import { PrintoutPanel } from './view/PrintoutPanel';
+import { showSidePanel } from 'oskari-ui/components/window';
+import { SIZE_OPTIONS, FORMAT_OPTIONS, PARAMS } from './constants';
+
+const BUNDLE_KEY = 'Printout';
+
+class UIHandler extends StateHandler {
+    constructor (instance) {
+        super();
+        this.instance = instance;
+        this.sandbox = Oskari.getSandbox();
+        this.mapModule = this.sandbox.findRegisteredModuleInstance('MainMapModule');
+        this.setState({
+            size: 'A4',
+            format: 'application/pdf',
+            mapTitle: '',
+            showScale: true,
+            showDate: true,
+            previewImage: null,
+            scaleType: 'map',
+            scale: null,
+            showTimeSeriesDate: false,
+            isMapStateChanged: false
+        });
+        this.sidePanel = null;
+        this.timeseriesPlugin = Oskari.getSandbox().findRegisteredModuleInstance('MainMapModuleTimeseriesControlPlugin');
+        this.eventHandlers = this.createEventHandlers();
+        this.scaleOptions = this.instance.conf?.scales || this.mapModule?.getScaleArray()?.slice().reverse();
+    };
+
+    showPanel () {
+        if (!this.sidePanel) {
+            this.sidePanel = showSidePanel(
+                <Message bundleKey={BUNDLE_KEY} messageKey='BasicView.title' />,
+                <PrintoutPanel
+                    controller={this.controller}
+                    state={this.state}
+                    scaleSelection={this.instance.conf.scaleSelection}
+                    scaleOptions={this.scaleOptions}
+                    isTimeSeries={this.isTimeSeriesActive()}
+                />,
+                () => this.closePanel()
+            );
+        }
+    }
+
+    updatePanel () {
+        if (this.sidePanel) {
+            this.sidePanel.update(
+                <Message bundleKey={BUNDLE_KEY} messageKey='BasicView.title' />,
+                <PrintoutPanel
+                    controller={this.controller}
+                    state={this.state}
+                    scaleSelection={this.instance.conf.scaleSelection}
+                    scaleOptions={this.scaleOptions}
+                    isTimeSeries={this.isTimeSeriesActive()}
+                />
+            );
+        }
+    }
```

Comment:
Try refactoring this in a way that there is no JSX used in the handler -> move it to instance and trigger calls to update with state change listeners provided by the StateHandler base class or something like that.